### PR TITLE
[RLlib] Fix ReplayBuffer doctest

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

While going through other issues, I found that the replay buffer doctest is broken.
it does not fail but a) everything is skipped and b) it clutters the log like this.

Screenshot 2023-02-12 at 20 42 31

### CODE REVIEW

**Review by sven1977 — APPROVED** [review]
LGTM. Thanks for the cleanup @ArturNiederfahrenhorst !

**Review by maxpumperla — APPROVED** [review]

File: rllib/utils/replay_buffers/replay_buffer.py [hunk 1]
```diff
@@ -77,46 +77,44 @@ class ReplayBuffer(ParallelIteratorWorker, FaultAwareApply):
...
+        print(buffer._eviction_started)
+        buffer.sample(1)
+
+        buffer = ReplayBuffer(capacity=10, storage_unit=StorageUnit.SEQUENCES)
+        buffer.add(SampleBatch({"c": [1, 2, 3], SampleBatch.SEQ_LENS: [1, 2]}))
+        buffer.sample(1)
+
+    .. testoutput::
+
+        True
```
Comment (gjoliver) [inline rllib/utils/replay_buffers/replay_buffer.py:116]: what is this ``True`` for?

Comment (ArturNiederfahrenhorst) [reply rllib/utils/replay_buffers/replay_buffer.py]: Screenshot 2023-02-14 at 21 34 18

For reference

File: rllib/utils/replay_buffers/replay_buffer.py [hunk 2]
```diff
@@ -77,46 +77,47 @@ class ReplayBuffer(ParallelIteratorWorker, FaultAwareApply):
...
+
+    `True` is not the output of the above testcode, but an artifact of unexpected
+    behaviour of sphinx doctests.
+    (see https://github.com/ray-project/ray/pull/32477#discussion_r1106776101)
```
Comment (ArturNiederfahrenhorst) [inline rllib/utils/replay_buffers/replay_buffer.py:120]: @gjoliver I've now added this explanation here for anyone stumbling across it.

Comment (gjoliver) [reply rllib/utils/replay_buffers/replay_buffer.py:120]: I see. thanks.

**Review by gjoliver — APPROVED** [review]

**Pull request merged**